axelor-account/src/main/java/com/axelor/apps/account/web/InvoiceController.java

```diff
@@ -500,18 +500,22 @@ public void showInvoice(ActionRequest request, ActionResponse response) {
                 ? Integer.parseInt(context.get("reportType").toString())
                 : null;
 
-        Map languageMap =
+        Map localizationMap =
             reportType != null
                     && (reportType == 1 || reportType == 3)
-                    && context.get("language") != null
-                ? (Map<String, Object>) request.getContext().get("language")
+                    && context.get("localization") != null
+                ? (Map<String, Object>) request.getContext().get("localization")
                 : null;
-        String locale =
-            languageMap != null && languageMap.get("id") != null
-                ? Beans.get(LanguageRepository.class)
-                    .find(Long.parseLong(languageMap.get("id").toString()))
+        String localizationCode =
+            localizationMap != null && localizationMap.get("id") != null
+                ? Beans.get(LocalizationRepository.class)
+                    .find(Long.parseLong(localizationMap.get("id").toString()))
                     .getCode()
                 : null;
+        String locale = "en";
+        if (localizationCode != null) {
+          locale = localizationCode.substring(0, 2);
+        }
```

Comment:
Just remove this code
We need the full locale and don't put a default value if it's null, if it's null it's already managed in InvoicePrintService.printInvoice method